autobahn/autobahn/wamp/protocol.py

```diff
@@ -481,15 +478,20 @@ def error(err):
                     raise ProtocolError("PUBLISHED received for non-pending request ID {0}".format(msg.request))
 
             elif isinstance(msg, message.Subscribed):
-
                 if msg.request in self._subscribe_reqs:
                     d, obj, fn, topic, options = self._subscribe_reqs.pop(msg.request)
-                    if options:
-                        self._subscriptions[msg.subscription] = Handler(obj, fn, topic, options.details_arg)
-                    else:
-                        self._subscriptions[msg.subscription] = Handler(obj, fn, topic)
-                    s = Subscription(self, msg.subscription)
-                    self._resolve_future(d, s)
+                    sub_id = msg.subscription
+                    # we should NEVER have our ID subscribed already
```

Comment:
No. We can receive the same subscription ID - if we are already subscribed. E.g. sending two `SUBSCRIBE` messages with same topic is fine. Router will answer with same subscription ID.